### PR TITLE
[IMP] warning: If a warning is set on a partner that is a Company, the w...

### PR DESCRIPTION
**Pull request by jat-odoo**
...arning also occur for the Contacts of this company
- Task: https://www.odoo.com/web#id=12551&view_type=form&model=project.task&menu_id=3942&action=327
- Pad: https://pad.odoo.com/p/r.JJu0JukaTooHoDwb

### CODE REVIEW

File: addons/warning/warning.py [hunk 1]
```diff
@@ -61,6 +61,12 @@ def onchange_partner_id(self, cr, uid, ids, part, context=None):
         message = False
         partner = self.pool.get('res.partner').browse(cr, uid, part, context=context)
         if partner.sale_warn != 'no-message':
+            partner = partner
```
Comment (rga-odoo) [inline addons/warning/warning.py]: it doesn't make sense. optimize your code

**Comment by xmo-odoo** [comment]
Wouldn't it make sense to have the whole "get the first ancestor with [attribute] != no-message" be cleanly bundled into a method?

File: addons/warning/warning.py (hunk 1 above)
Comment (jat-odoo) [inline addons/warning/warning.py]: @rga-odoo Done!

**Comment by hmo-odoo** [comment]
new PR: https://github.com/odoo/odoo/pull/7279

**Pull request closed without merging**